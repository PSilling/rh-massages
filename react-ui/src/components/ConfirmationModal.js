// react imports
import React, { Component } from 'react';

// component imports
import ModalActions from './ModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom confirmation modal component.
 */
class ConfirmationModal extends Component {

  render() {
    return(
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog onClose={this.props.onClose} width="40%">
            <h3>{ _t.translate('Action confirmation') }</h3>
            <p>{this.props.message}</p>
            <ModalActions
              primaryLabel={ _t.translate('Proceed') }
              onProceed={this.props.onConfirm}
              onClose={this.props.onClose}
            />
          </ModalDialog>
      </ModalContainer>
    )
  }
}

export default ConfirmationModal
