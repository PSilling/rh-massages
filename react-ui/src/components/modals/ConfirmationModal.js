// react imports
import React, { Component } from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";

// module imports
import { ModalContainer, ModalDialog } from "react-modal-dialog";

// component imports
import ModalActions from "../buttons/ModalActions";

// util imports
import _t from "../../util/Translations";

/**
 * Modal with a simple action confirmation message and ModalActions.
 */
class ConfirmationModal extends Component {
  handleKeyPress = event => {
    if (event.charCode === 13 && document.activeElement === ReactDOM.findDOMNode(this.modal)) {
      this.props.onConfirm();
    }
  };

  renderInsides = () => (
    <div>
      <h3>{_t.translate("Action confirmation")}</h3>
      <p>{this.props.message}</p>
      <ModalActions onProceed={this.props.onConfirm} onClose={this.props.onClose} autoFocus>
        {this.props.children}
      </ModalActions>
    </div>
  );

  render() {
    return this.props.withPortal ? (
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog
          onClose={this.props.onClose}
          width="40%"
          style={{ outline: "none" }}
          tabIndex="-1"
          onKeyPress={this.handleKeyPress}
          ref={dialog => {
            this.modalDialog = dialog;
          }}
        >
          {this.renderInsides()}
        </ModalDialog>
      </ModalContainer>
    ) : (
      this.renderInsides()
    );
  }
}

ConfirmationModal.propTypes = {
  /** children supplied to ModalActions */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /** message in the Modal */
  message: PropTypes.string.isRequired,
  /** callback function triggered on Modal close */
  onClose: PropTypes.func.isRequired,
  /** callback function triggered on primary button click */
  onConfirm: PropTypes.func.isRequired,
  /** whether ModalContainer should be used; useful for testing to avoid portals */
  withPortal: PropTypes.bool
};

ConfirmationModal.defaultProps = {
  children: null,
  withPortal: true
};

export default ConfirmationModal;
