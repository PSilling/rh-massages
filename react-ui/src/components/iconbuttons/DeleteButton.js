// react imports
import React, { Component } from 'react';

// component imports
import ConfirmationModal from '../modals/ConfirmationModal';

// util imports
import _t from '../../util/Translations';

/**
 * Custom delete button component.
 */
class DeleteButton extends Component {

  state = {active: false}

  handleToggle = () => {
    this.setState({active: !this.state.active});
  }

  render() {
    return(
      <span>
        <button style={{ 'color': '#000' }} type="button" className="btn btn-link"
          onClick={this.handleToggle} title={ _t.translate("Delete") }>
          <span className="glyphicon glyphicon-remove"></span>
        </button>
        {this.state.active ?
          <ConfirmationModal
            message={ _t.translate('Are you sure? This action cannot be reverted.') }
            onClose={() => this.handleToggle()}
            onConfirm={() => {
              this.handleToggle();
              this.props.onDelete();
            }}
          /> : ''
        }
      </span>
    )
  }
}

export default DeleteButton
