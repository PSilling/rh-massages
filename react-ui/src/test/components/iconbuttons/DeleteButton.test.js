// react imports
import React from 'react';
import TestRenderer from 'react-test-renderer';

// test imports
import DeleteButton from '../../../components/iconbuttons/DeleteButton';
import ConfirmationModal from '../../../components/modals/ConfirmationModal';

test('renders content with correct props', () => {
  const testFunction = jest.fn(),
        testRenderer = TestRenderer.create(
          <DeleteButton onDelete={testFunction} />
        ),
        testInstance = testRenderer.root;

  let button = testInstance.findByType('button'),
      icon = testInstance.findByProps({ className: "glyphicon glyphicon-remove" });

  button.props.onClick();

  let dialog = testInstance.findByType(ConfirmationModal),
      treeJSON = testRenderer.toJSON();

  expect(testFunction).not.toHaveBeenCalled();

  dialog.props.onConfirm();

  expect(testInstance.instance.state.active).toBe(false);
  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(treeJSON).toMatchSnapshot();
});
