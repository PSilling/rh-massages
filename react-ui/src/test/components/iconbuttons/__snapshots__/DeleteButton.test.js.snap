// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
<span>
  <button
    className="btn btn-link"
    onClick={[Function]}
    style={
      Object {
        "color": "#000",
      }
    }
    title="Smazat"
    type="button"
  >
    <span
      className="glyphicon glyphicon-trash"
    />
  </button>
</span>
`;
